File: 프로그래머스/lv1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        int temp = x;   // x값 임의 저장
        int compare = 0;
        while (temp > 0) {
            compare += temp % 10;
            temp /= 10;
        }
        if (x % compare == 0) {
            answer = true;
        } else {
            answer = false;
        }
        return answer;
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cs
public class Solution {
    public bool solution(int x) {
        bool answer = true;
        int sum = 0;
        int copyX = x;
        
        while (copyX > 0) {
            sum += copyX % 10;
            copyX /= 10;
        }
        
        if (x % sum == 0) {
            answer = true;
        } else {
            answer = false;
        }
        
        return answer;
    }
}

/*
 * 하샤드 수
 * x의 자릿수의 합을 구하려면 while문을 통해 x가 0이 될 때까지 반복하여 합을 구함.
 * 단, 값을 복사하여 사용
 * 나누어 떨어지면 true, 아니면 false
*/